### PR TITLE
API change: Change BrowserFeatureStateReporterPlugin signature

### PR DESCRIPTION
**Pull request by 0nko**
Task/Issue URL: https://app.asana.com/0/1202552961248957/1208307454923082/f

### Description

This PR changes the signature of `BrowserFeatureStateReporterPlugin` to be more flexible and return a map of key-value pairs.

### CODE REVIEW

**Comment by 0nko** [comment]
* **#5023** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5023?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @0nko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: common/common-utils/src/main/java/com/duckduckgo/common/utils/extensions/StringExtensions.kt [hunk 1]
```diff
@@ -87,3 +87,5 @@ fun String.compareSemanticVersion(targetVersion: String): Int? {
     }
     return 0
 }
+
+fun Boolean.toBinaryString(): String = if (this) "1" else "0"
```
Comment (mikescamell) [inline common/common-utils/src/main/java/com/duckduckgo/common/utils/extensions/StringExtensions.kt:91]: 🔍 I think this is probably more appropriate in a `BooleanExtensions` file

**Review by mikescamell — APPROVED** [review]
Nice work! 🥳 Looks fine to me but probably worth waiting for someone with more codebase experience to check 👍

File: common/common-utils/src/main/java/com/duckduckgo/common/utils/extensions/StringExtensions.kt (hunk 1 above)
Comment (0nko) [reply common/common-utils/src/main/java/com/duckduckgo/common/utils/extensions/StringExtensions.kt]: You’re right.. Fixed!

**Pull request merged**